buildkite/webhook_test.go

```diff
@@ -0,0 +1,166 @@
+	tests := []struct {
+		payload      interface{}
+		messageType  string
+		errorMessage string
+	}{
+		{
+			payload:     &AgentConnectedEvent{},
+			messageType: "agent.connected",
+		},
+		{
+			payload:     &AgentDisconnectedEvent{},
+			messageType: "agent.disconnected",
+		},
+		{
+			payload:     &AgentLostEvent{},
+			messageType: "agent.lost",
+		},
+		{
+			payload:     &AgentStoppedEvent{},
+			messageType: "agent.stopped",
+		},
+		{
+			payload:     &AgentStoppingEvent{},
+			messageType: "agent.stopping",
+		},
+		{
+			payload:     &BuildFinishedEvent{},
+			messageType: "build.finished",
+		},
+		{
+			payload:     &BuildRunningEvent{},
+			messageType: "build.running",
+		},
+		{
+			payload:     &BuildScheduledEvent{},
+			messageType: "build.scheduled",
+		},
+		{
+			payload:     &JobActivatedEvent{},
+			messageType: "job.activated",
+		},
+		{
+			payload:     &JobFinishedEvent{},
+			messageType: "job.finished",
+		},
+		{
+			payload:     &JobScheduledEvent{},
+			messageType: "job.scheduled",
+		},
+		{
+			payload:     &JobStartedEvent{},
+			messageType: "job.started",
+		},
+		{
+			payload:     &PingEvent{},
+			messageType: "ping",
+		},
+		{
+			payload:      &PingEvent{},
+			messageType:  "invalid",
+			errorMessage: "unknown X-Buildkite-Event in message: invalid",
+		},
+	}
+
+	for _, test := range tests {
+		p, err := json.Marshal(test.payload)
+		if err != nil {
+			t.Fatalf("Marshal(%#v): %v", test.payload, err)
+		}
+
+		got, err := ParseWebHook(test.messageType, p)
+		if err != nil {
+			if test.errorMessage != "" {
+				if err.Error() != test.errorMessage {
+					t.Errorf("ParseWebHook(%#v, %#v) expected error, got %#v", test.messageType, test.payload, err.Error())
+				}
+				continue
+			}
+			t.Fatalf("ParseWebHook: %v", err)
+		}
+
+		if want := test.payload; !reflect.DeepEqual(got, want) {
+			t.Errorf("ParseWebHook(%#v, %#v) = %#v, want %#v", test.messageType, p, got, want)
+		}
+	}
+}
+
+func TestValidatePayload(t *testing.T) {
+	const defaultBody = `{"event":"ping","service":{"id":"c9f8372d-c0cd-43dc-9274-768a875cf6ca","provider":"webhook","settings":{"url":"https://server.com/webhooks"}},"organization":{"id":"49801950-1df0-474f-bb56-ad6a930c5cb9","graphql_id":"T3JnYW5pemF0aW9uLS0tZTBmMzk3MgsTksGkxOWYtZTZjNzczZTJiYjEy","url":"https://api.buildkite.com/v2/organizations/acme-inc","web_url":"https://buildkite.com/acme-inc","name":"ACME Inc","slug":"acme-inc","agents_url":"https://api.buildkite.com/v2/organizations/acme-inc/agents","emojis_url":"https://api.buildkite.com/v2/organizations/acme-inc/emojis","created_at":"2021-02-03T20:34:10.486Z","pipelines_url":"https://api.buildkite.com/v2/organizations/acme-inc/pipelines"},"sender":{"id":"c9f8372d-c0cd-43dc-9269-bcbb7f308e3f","name":"ACME Man"}}`
+	const defaultSignature = "timestamp=1642080837,signature=582d496ac2d869dd97a3101c4cda346288c49a742592daf582ec64c86449f79c"
+	const payloadSignatureError = "payload signature check failed"
```

Comment:
It doesn't look like this value `payloadSignatureError` is used anywhere, was intended to be referenced somewhere?

Reply:
Good spot @jeremybumsted ... I _did_ mean to use it.

I've updated the testing to be more explicit about the errors being returned in [`0e335ed` (#97)](https://github.com/buildkite/go-buildkite/pull/97/commits/0e335ed7352a4392036bd8ebda67ce8a42b352ae).

PTAL